power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -714,50 +766,113 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
   public:
     template <class Component> using UpdateType = typename Component::UpdateType;
 
-    static bool is_update_independent(ConstDataset const& update_data) {
-        // If the batch size is (0 or) 1, then the update data for this component is 'independent'
-        if (update_data.batch_size() <= 1) {
-            return true;
-        }
+    static UpdateCompIndependence check_components_independence(ConstDataset const& update_data) {
```

Comment:
with the amount of checks happening here, i'm also in favor of moving these `static` functions out of main model. feel free to do that in this PR. if you prefer to keep it, that's also OK

Reply:
Refactor scoped out. Noted in https://github.com/PowerGridModel/power-grid-model/issues/806